CRM/Activityreport/Data.php

```diff
@@ -15,65 +16,119 @@ class CRM_Activityreport_Data {
    * 
-   * @param array   $data       array containing a set of Activities
+   * @param array   $data               array containing a set of Activities
+   * @param int     $totalOffset        Activity absolute offset we start with
+   * @param int     $multiValuesOffset  Multi Values offset
    * 
    * @return array
    */
-  protected static function splitMultiValues(array $data) {
+  protected static function splitMultiValues(array $data, $totalOffset, $multiValuesOffset) {
     $result = array();
+    $messages = array();
+    $i = 0;
 
     foreach ($data as $key => $row) {
+      $multiValuesRows = array();
       if (!empty(self::$multiValues[$key])) {
         $multiValuesFields = array_combine(self::$multiValues[$key], array_fill(0, count(self::$multiValues[$key]), 0));
-        $result = array_merge($result, self::populateMultiValuesRow($row, $multiValuesFields));
+        $multiValuesRows = self::populateMultiValuesRow($row, $multiValuesFields, $multiValuesOffset, self::ROWS_TO_RETURN - $i);
+        $result = array_merge($result, $multiValuesRows['data']);
+        $messages = array_merge($messages, $multiValuesRows['info']['messages']);
+        $multiValuesOffset = 0;
       } else {
         $result[] = $row;
       }
+      $i = count($result);
+      if ($i === self::ROWS_TO_RETURN) {
+        break;
+      }
+      $totalOffset++;
     }
 
-    return $result;
+    $header = self::getHeader();
+    $output = array(array_keys($header));
+    foreach ($result as $row) {
+      $output[] = array_values($row);
+    }
+
+    return array(
+      array(
+        'info' => array(
+          'nextOffset' => !empty($multiValuesRows['info']['multiValuesOffset']) ? $totalOffset : $totalOffset + 1,
+          'multiValuesOffset' => !empty($multiValuesRows['info']['multiValuesOffset']) ? $multiValuesRows['info']['multiValuesOffset'] : 0,
+          'multiValuesTotal' => !empty($multiValuesRows['info']['multiValuesTotal']) ? $multiValuesRows['info']['multiValuesTotal'] : 0,
+          'messages' => $messages,
+        ),
+        'data' => $output,
+      ),
+    );
+  }
+
+  protected static function getHeader() {
+    $header = array_merge(self::$emptyRow, array(
+      'Activity Date' => null,
+      'Activity Start Date Months' => null,
+      'Activity is a test' => null,
+      'Activity Expire Date' => null,
+    ));
+    ksort($header);
+    return $header;
   }
 
   /**
    * Return an array containing set of rows which are built basing on given $row
    * and $fields array with indexes of multi values of the $row.
    * 
-   * @param array   $row        a single Activity row
-   * @param array   $fields     array containing Activity multi value fields
-   *                            as keys and integer indexes as values
+   * @param array   $row          a single Activity row
+   * @param array   $fields       array containing Activity multi value fields
+   *                              as keys and integer indexes as values
```

Comment:
please remove this space , also check my comment about drupal standrad for docblocks